spec/models/ride_spec.rb

```diff
@@ -3,5 +3,45 @@
 RSpec.describe Ride, type: :model do
   describe "relationships" do
     it { should belong_to(:amusement_park) }
+    it { should have_many(:mechanic_rides) }
+    it { should have_many(:mechanics).through(:mechanic_rides) }
+  end
+
+  describe "validations" do
+    it { should validate_presence_of :name }
+    it { should validate_presence_of :thrill_rating }
+    it { should allow_value(true).for(:open) }
+    it { should allow_value(false).for(:open) }
+  end
+
+  before :each do
+    @sharon = Mechanic.create!(name: "Sharon", years_experience: 12)
+    @maria = Mechanic.create!(name: "Maria", years_experience: 20)
+    @geraldine = Mechanic.create!(name: "Geraldine", years_experience: 4)
+    
+    @elitch_gardens = AmusementPark.create!(name: "Elitch Gardens", admission_cost: 65)
+    
+    @twister = @elitch_gardens.rides.create!(name: "Twister", thrill_rating: 6, open: true)
+    @carousel = @elitch_gardens.rides.create!(name: "Carousel", thrill_rating: 2, open: true)
+    @dragon_force = @elitch_gardens.rides.create!(name: "Dragon Force", thrill_rating: 10, open: true)
+
+    MechanicRide.create!(mechanic: @sharon, ride: @twister)
+    MechanicRide.create!(mechanic: @maria, ride: @twister)
+    MechanicRide.create!(mechanic: @sharon, ride: @carousel)
+    MechanicRide.create!(mechanic: @geraldine, ride: @carousel)
+    MechanicRide.create!(mechanic: @maria, ride: @dragon_force)
+    MechanicRide.create!(mechanic: @geraldine, ride: @dragon_force)
+  end
+
+  describe "avg_mechanic_experience" do
+    it "returns the average number of years of experience between all mechanics working on a ride" do
```

Comment:
For this method, we actually want to get back an array of the rides, and each ride should have a virtual attribute with the avg years of experience (created from the query with the aggregate function). So you should have an assertion that verifies they order they're returned in, and then for each ride in the array, you can have an assertion calling on that virtual attribute to make sure the aggregate calculation is correct.